lib/auth/oidc/client_assertion.go

```diff
@@ -195,3 +204,22 @@ func hashKeyID(cert *x509.Certificate, header structs.OIDCClientAssertionKeyIDHe
 	hashed := hasher.Sum(nil)
 	return base64.RawURLEncoding.EncodeToString(hashed), nil
 }
+
+var (
+	// for user convenience, they may exclude key/cert pem header/footer
+	beginPrivateKey  = []byte("-----BEGIN PRIVATE KEY-----")
+	endPrivateKey    = []byte("-----END PRIVATE KEY-----")
+	beginCertificate = []byte("-----BEGIN CERTIFICATE-----")
+	endCertificate   = []byte("-----END CERTIFICATE-----")
+)
+
+// wrapBeginEnd wraps the provided bts in "{begin}\n{bts}\n{end}".
+// if begin and end are already there, it ensures "\n" is between them and bts.
+func wrapBeginEnd(bts, begin, end []byte) []byte {
+	cp := bytes.Clone(bts)
+	cp = bytes.TrimPrefix(cp, begin)
```

Comment:
We might want to trim spaces here too, so that we can handle leading/trailing newlines nicely:

```suggestion
	cp = bytes.TrimSpace(cp)
	cp = bytes.TrimPrefix(cp, begin)
```

Reply:
It's funny (frustrating) how finicky it is - newlines before/after don't seem to matter! You can even put total nonsense in there, so long as it's outside the header/footer, e.g. this is valid:

```
ignore this
bogus junk
-----BEGIN PRIVATE KEY-----
[...pretend the key material is in here...]
-----END PRIVATE KEY-----
again
more nonsense!
```

We can still trim it though, if the uniformity is worth the extra string manipulation (these calls shouldn't happen toooooo often).